javascript/organizing_your_javascript_code/objects_and_object_constructors.md

```diff
@@ -149,94 +149,204 @@ console.log(theHobbit.info());
+2. **The prototype is another object**: 
+
+   2a. The _value_ of the Object Constructor's `.prototype` property (i.e., `Player.prototype`) contains the `prototype` object. 
+   
+   2b. The _reference_ to this value of `Player.prototype` is stored in every `Player` object's `.__proto__` property, every time a `Player` object is created. 
+   
+   2c. Hence, you get a `true` value returned when you compute `player1.__proto__ === Player.prototype`.
+
+3. **...that the original object _inherits_ from, and has access to all of its prototype's methods and properties**: 
+
+   3a. As said in the earlier point, every `Player` object's `.__proto__` property refers to `Player.prototype`: `player1.__proto__ === player2.__proto__` (returns `true`).
+   
+   3b. So, any properties or methods defined on `Player.prototype` will be available to the created `Player` objects!
+   
+Point 3b needs a little more explanation. What does defining 'on the `prototype`' mean? Consider the following code:
+
+~~~javascript
+Player.prototype.sayHello = function() {
+   console.log("Hello, I'm a player!");
 }
+
+player1.sayHello() // logs "Hello, I'm a player!"
+player2.sayHello() // logs "Hello, I'm a player!"
+~~~
+
+Here, we defined the `.sayHello` function 'on' the `Player.prototype` object. It then became available for the `player1` and the `player2` objects to use! Similarly, you can attach other properties or functions you want to use on all `Player` objects by defining them on the objects' prototype (`Player.prototype`).
+
+#### Object.getPrototypeOf() vs. .__proto__ vs. [[Prototype]]
+
+As you have seen above, accessing an object's `prototype` was done using the `.__proto__` property of the object. However, this is a non-standard way of doing so, and it is recommended to access an object's `prototype` by using the [`Object.getPrototypeOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getPrototypeOf) function. The same code can thus be rewritten to become:
+
+~~~javascript
+Object.getPrototypeOf(player1) === Player.prototype // returns true
+Object.getPrototypeOf(player2) === Player.prototype // returns true
 ~~~
 
-If you're using constructors to make your objects it is best to define functions on the `prototype` of that object. Doing so means that a single instance of each function will be shared between all of the Student objects. If we declare the function directly in the constructor, like we did when they were first introduced, that function would be duplicated every time a new Student is created. In this example, that wouldn't really matter much, but in a project that is creating thousands of objects, it really can make a difference.
+In some places you might also come across `[[Prototype]]`, which is just another way of talking about the `.__proto__` property of an object, like `player1.[[Prototype]]`.
+
+This explanation about the `prototype` might have been a lot, so remember to take a breather before moving on!
```

Comment:
Use this section to make a note about `__proto__` and `[[Prototype]]`, mention how you might see them in legacy code, etc... See the bigger comment on the review to understand why we prefer not teaching to use `__proto__` in lieu of `Object.getPrototypeOf()`